### PR TITLE
Add functionality for BBspec `get_nl`

### PR DESCRIPTION
**Pull request by CheukHinHoJerry**
ref : #204

### CODE REVIEW

**Comment by CheukHinHoJerry** [comment]
I think we don't really need to store the `BBspec` in the model, given that we didn't do that (as far as I know) for previous `acemodel` too.

File: src/models/sparse.jl [hunk 1]
```diff
@@ -57,3 +57,21 @@ function pullback_evaluate(∂B, tensor::SparseEquivTensor{T}, Rnl, Ylm,
    return ∂Rnl, ∂Ylm
 end
 
+
+_nl(bb) = [(n = b.n, l = b.l) for b in bb]
```
Comment (cortner) [inline src/models/sparse.jl]: maybe move this *inside* `get_nl`

File: src/models/sparse.jl [hunk 2]
```diff
@@ -57,3 +57,21 @@ function pullback_evaluate(∂B, tensor::SparseEquivTensor{T}, Rnl, Ylm,
...
+
+function get_nl(tensor::SparseEquivTensor{T}) where {T}
+   # assume the new ACE model NEVER has the z channel
```
Comment (cortner) [inline src/models/sparse.jl:70]: yes, I think we want to guarantee *HERE* at least that there are only `(n, l, m)` channels. If that ever changes then we will of course have to revisit.

File: src/models/sparse.jl [hunk 3]
```diff
@@ -57,3 +57,21 @@ function pullback_evaluate(∂B, tensor::SparseEquivTensor{T}, Rnl, Ylm,
    return ∂Rnl, ∂Ylm
 end
 
+
+_nl(bb) = [(n = b.n, l = b.l) for b in bb]
+
+function get_nl(tensor::SparseEquivTensor{T}) where {T}
```
Comment (cortner) [inline src/models/sparse.jl]: can we call this `get_nnll_spec` please? Can can you add a short inline documentation also to explain what this returns?

**Review by cortner — CHANGES_REQUESTED** [review]
Thank you. This looks good, can you please just take care of the minor issues I pointed out? 

I'm not sure how to write tests for this, so let's just not worry about it, cross our fingers and hope for the best?!? If anybody has an idea then we should of course add a unit test here.

File: src/models/sparse.jl (hunk 3 above)
Comment (CheukHinHoJerry) [reply src/models/sparse.jl]: The reason I call this `get_nl` is that `ACE1` and `ACEpotentials` calls that too. But this is a new model type so we can probably not worry about it?

**Review by cortner — APPROVED** [review]

File: src/models/sparse.jl (hunk 3 above)
Comment (cortner) [reply src/models/sparse.jl]: I see - I prefer we don't tie ourselves to bad decisions made in the past :)

**Comment by cortner** [comment]
happy to merge?

**Comment by CheukHinHoJerry** [comment]
Yes please.

**Pull request merged**